### PR TITLE
Add the HELLO command response when creating a connection

### PR DESCRIPTION
**Pull request by swelham**
Todo
* [x] Complete connection setup by sending the `HELLO` command
* [x] Complete tests on all new public functions
* [x] Make sure error cases are tested

### CODE REVIEW

**Comment by swelham** [comment]
[Faktory low level client](https://trello.com/c/eNOykYB2/1077-faktory-low-level-client)

**Comment by swelham** [comment]
@SeatedInc/dev This is ready for review

File: lib/faktory_worker/connection.ex [hunk 1]
```diff
@@ -3,17 +3,60 @@ defmodule FaktoryWorker.Connection do
...
+  end
+
+  defp verify_version({:ok, %{"v" => 2}}), do: :ok
+
+  defp verify_version({:ok, %{"v" => version}}) do
+    {:error, "Only Faktory version '2' is supported (connected to Faktory version '#{version}')."}
+  end
+
+  defp verify_hello_response(connection) do
+    with :ok <- send_command(connection, {:hello, 2}),
```
Comment (beerlington) [inline lib/faktory_worker/connection.ex]: I feel like `2` should be extracted to a named module attribute like `@faktory_version`. I wasn't 100% sure what `2` meant here until I checked `Protocol.encode_command/1`.

Comment (swelham) [reply lib/faktory_worker/connection.ex]: Thanks @beerlington, now updated.

**Pull request merged**